### PR TITLE
Update VPA CRD objects' statuses

### PR DESCRIPTION
**Pull request by schylek**
* Allow patch on VPA CRD objects instead of put (to avoid race condition with user actions).
* Store VPA and Pod info in ClusterState.
* Update VPA status fields with recommendations for containers.
* Minor fixes.

### CODE REVIEW

File: vertical-pod-autoscaler/recommender/recommender.go [hunk 1]
```diff
@@ -53,6 +58,10 @@ type recommender struct {
 	podResourceRecommender  logic.PodResourceRecommender
 }
 
+type verticalPodAutoscalerStatusPatch struct {
+	Status vpa_types.VerticalPodAutoscalerStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/recommender.go]: I don't that is the intended mechanism for patching ...

**Comment by schylek** [comment]
Changed update method from JSON merge to JSON patch (https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#patch-operations). The API is a little bit cumbersome, since Status object inside VPA object has to exist before it can be patched and there is no way to atomically solve this - call to initialize status may actually override someone else's update, but the recommender should be the first one to fill the Status anyway.

**Comment by mwielgus** [comment]
@schylek  - Please resolve conflicts.

**Comment by schylek** [comment]
Conflicts resolved.

**Pull request merged**